modules/javafx.graphics/src/main/java/javafx/scene/Node.java

```diff
@@ -8305,6 +8311,19 @@ protected PseudoClass getPseudoClass() {
         public String getName() {
             return "focusWithin";
         }
+
+        /**
+         * Adjusts the number of focused nodes within this node.
+         */
+        void adjust(int change) {
+            count += change;
+
+            if (count == 1) {
+                set(true);
+            } else if (count == 0) {
+                set(false);
+            }
```

Comment:
On the one hand, this might make the code a little bit more robust, but on the other hand, it might hide a bug elsewhere. Surely `count` should never be negative. I lean slightly towards not protecting code against bugs in this way, mostly because it might expose potential bugs sooner.

Reply:
I've created a ticket to investigate this: https://bugs.openjdk.org/browse/JDK-8301556